mypy/fastparse.py

```diff
@@ -596,7 +596,7 @@ def fix_function_overloads(self, stmts: list[Statement]) -> list[Statement]:
             if_overload_name: str | None = None
             if_block_with_overload: Block | None = None
             if_unknown_truth_value: IfStmt | None = None
-            if isinstance(stmt, IfStmt) and seen_unconditional_func_def is False:
+            if isinstance(stmt, IfStmt) and not seen_unconditional_func_def:
```

Comment:
This one is questionable. I would prefer the explicit `is False`. Especially in combination with `and` / `or` it's much to easy to make mistakes with `not`.

-> Revert `FURB149`